doc/ikfast/ikfast_tutorial.rst

```diff
@@ -73,210 +53,79 @@ Inside your catkin workspace's ``./src`` directory: ::
 
-To make using this tutorial copy/paste friendly, set a ``MYROBOT_NAME`` environment variable with the name of your robot: ::
+To facilitate copy-and-paste, we suggest to define the robot name as an environment variable: ::
 
- export MYROBOT_NAME="panda_arm"
+  export MYROBOT_NAME="panda"
 
-First you will need robot description file that is in `Collada or OpenRAVE <http://openrave.org/docs/latest_stable/collada_robot_extensions/>`_ robot format.
+OpenRAVE uses Collada instead of URDF to describe the robot. In order to automatically convert your robot's URDF to Collada, you need to provide the .urdf file.
+If your .urdf file is generated from `xacro <http://wiki.ros.org/xacro/>`_ files you can generate the urdf using the following command: ::
 
-If your robot is not in this format we recommend you create a ROS `URDF <http://www.ros.org/wiki/urdf/Tutorials/Create%20your%20own%20urdf%20file>`_ file.
-
-If your robot is in `xacro <http://wiki.ros.org/xacro/>`_ format you can convert it to urdf using the following command: ::
-
- rosrun xacro xacro --inorder -o "$MYROBOT_NAME".urdf "$MYROBOT_NAME".urdf.xacro
-
-Once you have your robot in URDF format, you can convert it to Collada (.dae) file using the following command: ::
-
- rosrun collada_urdf urdf_to_collada "$MYROBOT_NAME".urdf "$MYROBOT_NAME".dae
-
-**Note:** you may need to install ``collada_urdf``: ::
-
- sudo apt install ros-${ROS_DISTRO}-collada-urdf
-
-Often floating point issues arise in converting a URDF file to Collada file, so a script has been created to round all the numbers down to x decimal places in your .dae file. Its probably best if you skip this step initially and see if IKFast can generate a solution with your default values, but if the generator takes longer than, say, an hour, try the following: ::
-
- export IKFAST_PRECISION="5"
- cp "$MYROBOT_NAME".dae "$MYROBOT_NAME".backup.dae  # create a backup of your full precision dae.
- rosrun moveit_kinematics round_collada_numbers.py "$MYROBOT_NAME".dae "$MYROBOT_NAME".dae "$IKFAST_PRECISION"
-
-From experience we recommend 5 decimal places, but if the OpenRAVE IKFast generator takes to long to find a solution, lowering the number of decimal places should help.
-
-To see the links in your newly generated Collada file
-
-You may need to install package **libsoqt4-dev** to have the display working: ::
-
- openrave-robot.py "$MYROBOT_NAME".dae --info links
-
-This is useful if you have a 7-dof arm and you need to fill in a ``--freeindex`` parameter, discussed later.
-
-To test your newly generated Collada file in OpenRAVE: ::
-
- openrave "$MYROBOT_NAME".dae
-
-You should see your robot.
-
-.. image:: openrave_panda.png
-   :width: 700px
-
-Create IKFast Solution CPP File
--------------------------------
-Once you have a numerically rounded Collada file its time to generate the C++ .h header file that contains the analytical IK solution for your robot.
+  rosrun xacro xacro -o $MYROBOT_NAME.urdf $MYROBOT_NAME.urdf.xacro
```

Comment:
This needs to be run in `catkin_ws/src`, because there the new package will be created (in the current working dir).
If your .urdf file resides somewhere else, you need to provide the full path. For that reason, I'm not a fan of using the `MYROBOT_NAME` variable...